### PR TITLE
docs: improve cleverMerge to respect byDependency order correctly

### LINKED ISSUE #3844: Document webpack change: Improve cleverMerge to respect byDependency order correctly
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/11160

---

* Allow to use "..." in `option.resolve`, `cache.managed/immutablePaths`, `module.defaultRules`, `optimization.minimizer` to access the defaults
* Add "..." to schema when needed
* Apply dependency specific mainFields, aliasFields and conditionNames only for cjs and esm dependencies or when none is specified for backward-compat.

**What kind of change does this PR introduce?**
bugfix, refactor

**Did you add tests for your changes?**
yes

**Does this PR introduce a breaking change?**
yes

**What needs to be documented once your changes are merged?**
* When using `this.getResolve` in loader, make sure to specify an appropriate `dependencyType`, e. g. `esm` or `commonjs` or a custom one.
* `this.loadModule` in loader context uses commonjs resolve rules by default. Use `this.getResolve` with `dependencyType` before to use a different semantic.
* `"..."` can be used in `option.resolve`, `cache.managed/immutablePaths`, `module.defaultRules`, `optimization.minimizer` to access the defaults, e. g. `optimization.mininizer: [new CssMinimizer(), "..."]`

### PR DESCRIPTION
**Pull request by jamesgeorge007**
fixes #3844

### CODE REVIEW

File: src/content/api/loaders.md [hunk 1]
```diff
@@ -389,6 +390,8 @@ loadModule(request: string, callback: function(err, source, sourceMap, module))
 
 Resolves the given request to a module, applies all configured loaders and calls back with the generated source, the sourceMap and the module instance (usually an instance of [`NormalModule`](https://github.com/webpack/webpack/blob/master/lib/NormalModule.js)). Use this function if you need to know the source code of another module to generate the result.
 
+`this.loadModule` in loader context uses commonjs resolve rules by default. Use `this.getResolve` with an appropriate `dependencyType`, e. g. `esm` or `commonjs` or a custom one before using a different semantic.
```
Comment (montogeek) [inline src/content/api/loaders.md:393]:
```suggestion
`this.loadModule` in a loader context uses CommonJS resolve rules by default. Use `this.getResolve` with an appropriate `dependencyType`, e. g. `esm` or `commonjs` or a custom one before using a different semantic.
```

File: src/content/api/loaders.md [hunk 2]
```diff
@@ -389,6 +390,8 @@ loadModule(request: string, callback: function(err, source, sourceMap, module))
 
 Resolves the given request to a module, applies all configured loaders and calls back with the generated source, the sourceMap and the module instance (usually an instance of [`NormalModule`](https://github.com/webpack/webpack/blob/master/lib/NormalModule.js)). Use this function if you need to know the source code of another module to generate the result.
 
+`this.loadModule` in a loader context uses CommonJS resolve rules by default. Use `this.getResolve` with an appropriate `dependencyType`, e. g. `esm` or `commonjs` or a custom one before using a different semantic.
```
Comment (EugeneHlushko) [inline src/content/api/loaders.md:393]:
```suggestion
`this.loadModule` in a loader context uses CommonJS resolve rules by default. Use `this.getResolve` with an appropriate `dependencyType`, e. g. `'esm'`, `'commonjs'` or a custom one before using a different semantic.
```

File: src/content/configuration/optimization.md [hunk 3]
```diff
@@ -48,6 +48,8 @@ T> Learn how [mode](/configuration/mode/) works.
 
 Allows you to override the default minimizer by providing a different one or more customized [TerserPlugin](/plugins/terser-webpack-plugin/) instances.
 
+`"..."` can be used in `optimization.minimizer` to access the defaults.
```
Comment (EugeneHlushko) [inline src/content/configuration/optimization.md:51]: This is asking for an example when i read it. What do you think if we move it to a place below existing examples and add one showcasing accessing the defaults via ...?

Comment (jamesgeorge007) [reply src/content/configuration/optimization.md:52]: Sounds good 👍

File: src/content/configuration/optimization.md [hunk 4]
```diff
@@ -85,6 +85,16 @@ module.exports = {
 };
 ```
 
+`"..."` can be used in `optimization.minimizer` to access the defaults.
```
Comment (EugeneHlushko) [inline src/content/configuration/optimization.md:88]:
```suggestion
`'...'` can be used in `optimization.minimizer` to access the defaults.
```

**Comment by EugeneHlushko** [comment]
Thanks!

**Pull request merged**